### PR TITLE
Feich/add more ut

### PR DESCRIPTION
**Pull request by feich-ms**
Add more unit tests. Current coverage rate is about 66%
## Description
1. add unit tests for Outlook service
2. add more corner case assertion for flow tests
3. fix a bug of todo ListType

## Related Issue
[en-US] VA - [Todo] "Add buy fruit" added to general list

## Checklist
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated the appropriate unit tests

### CODE REVIEW

File: solutions/Virtual-Assistant/src/csharp/skills/tests/todoskilltest/API/OutlookServiceTests.cs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+{
+    [TestClass]
+    public class OutlookServiceTests
+    {
+        private HttpClient mockClient;
+
+        [TestInitialize]
+        public void Initialize()
+        {
+            if (mockClient == null)
```
Comment (lzc850612) [inline solutions/Virtual-Assistant/src/csharp/skills/tests/todoskilltest/API/OutlookServiceTests.cs]: it's always gonna be null so no need to check :)

File: solutions/Virtual-Assistant/src/csharp/skills/tests/todoskilltest/Flow/Fakes/MockData.cs [hunk 2]
```diff
@@ -173,9 +179,23 @@ public static class MockData
...
+        public static string ImageSource { get; } = "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAABGdBTUEAALGPC/xhBQAAACBjSFJNAAB6JgAAgIQAAPoAAACA6AAAdTAAAOpgAAA6mAAAF3CculE8AAAACXBIWXMAAA7EAAAOxAGVKw4bAAAAAmJLR0QAAKqNIzIAAAAHdElNRQfiDAMKKhgxjYNuAAAAJXRFWHRkYXRlOmNyZWF0ZQAyMDE4LTEyLTAzVDEwOjQyOjI0KzAxOjAw3NT2SAAAACV0RVh0ZGF0ZTptb2RpZnkAMjAxOC0xMi0wM1QxMDo0MjoyNCswMTowMK2JTvQAAAAZdEVYdFNvZnR3YXJlAHd3dy5pbmtzY2FwZS5vcmeb7jwaAAAA/0lEQVQ4T6WTzQmEMBCFx0XQErypHXj0KliCPYgNiUcLsAev3uxAvVqComTzhrgu/i1hPxjykuG9SBJJSMIwFESkVfAAA6JpGoqiiEzTRKDs32MYBi3LQnVdk/TympBmTtMBHnhfSMDOumweDpCBPHliXVcax1HNdg8H/GKaJt7RcRy1svMzYJ5nsm2bte/7PH5zCijLknArAGbLsljD3HUd6yMijmM+WYA5qqqqj/Y8T3V34EHv9AVZlvGYJAmP0kx937O+4hSQ5zmlacradd1HM7g8xKIoqG1bGoZBrdzDAXieR4IgUOqazcMBeNu6bJ7/fyZcCUJkT6vgEUKINxqN2iFI/P1RAAAAAElFTkSuQmCC";
```
Comment (lzc850612) [inline solutions/Virtual-Assistant/src/csharp/skills/tests/todoskilltest/Flow/Fakes/MockData.cs]: can this be in a file? i really don't like having a long string like this in the code

File: solutions/Virtual-Assistant/src/csharp/skills/tests/todoskilltest/Flow/Fakes/MockData.cs [hunk 3]
```diff
@@ -7,7 +7,20 @@ namespace ToDoSkillTest.Flow.Fakes
 {
     public static class MockData
     {
-        public static List<TaskItem> MockTaskItems = new List<TaskItem>
+        public enum DataChangeType
```
Comment (lzc850612) [inline solutions/Virtual-Assistant/src/csharp/skills/tests/todoskilltest/Flow/Fakes/MockData.cs]: better to declare this in a separate class file

**Review by lzc850612 — CHANGES_REQUESTED** [review]

File: solutions/Virtual-Assistant/src/csharp/skills/tests/todoskilltest/API/OutlookServiceTests.cs (hunk 1 above)
Comment (feich-ms) [reply solutions/Virtual-Assistant/src/csharp/skills/tests/todoskilltest/API/OutlookServiceTests.cs]: removed

File: solutions/Virtual-Assistant/src/csharp/skills/tests/todoskilltest/Flow/Fakes/MockData.cs (hunk 2 above)
Comment (feich-ms) [reply solutions/Virtual-Assistant/src/csharp/skills/tests/todoskilltest/Flow/Fakes/MockData.cs]: removed and use the data from ToDo skill project

File: solutions/Virtual-Assistant/src/csharp/skills/tests/todoskilltest/Flow/Fakes/MockData.cs (hunk 3 above)
Comment (feich-ms) [reply solutions/Virtual-Assistant/src/csharp/skills/tests/todoskilltest/Flow/Fakes/MockData.cs]: good suggestion. Moved it to a separated file.

**Review by lzc850612 — APPROVED** [review]
